utils/results/conversion/convertor.go

```diff
@@ -0,0 +1,191 @@
+package conversion
+
+import (
+	"strings"
+
+	"github.com/jfrog/jfrog-cli-security/utils/formats"
+	"github.com/jfrog/jfrog-cli-security/utils/results"
+	"github.com/jfrog/jfrog-cli-security/utils/results/conversion/sarifparser"
+	"github.com/jfrog/jfrog-cli-security/utils/results/conversion/simplejsonparser"
+	"github.com/jfrog/jfrog-cli-security/utils/results/conversion/summaryparser"
+	"github.com/jfrog/jfrog-cli-security/utils/results/conversion/tableparser"
+	"github.com/jfrog/jfrog-cli-security/utils/techutils"
+	"github.com/jfrog/jfrog-client-go/xray/services"
+	"github.com/owenrumney/go-sarif/v2/sarif"
+)
+
+type CommandResultsConvertor struct {
+	Params ResultConvertParams
+}
+
+type ResultConvertParams struct {
+	// Control and override converting command results as multi target results
+	IsMultipleRoots *bool
+	// Control if the output should include licenses information
+	IncludeLicenses bool
+	// Control if the output should include vulnerabilities information
+	IncludeVulnerabilities bool
+	// Output will contain only the unique violations determined by the GetUniqueKey function
+	SimplifiedOutput bool
+	// Create local license violations if repo context was not provided and a license is not in this list
+	AllowedLicenses []string
+	// Convert the results to a pretty format if supported
+	Pretty bool
+	// Relevant for Sarif format since Github format does not support results without locations
+	AllowResultsWithoutLocations bool
+}
+
+func NewCommandResultsConvertor(params ResultConvertParams) *CommandResultsConvertor {
+	return &CommandResultsConvertor{Params: params}
+}
+
+// Parse a stream of results and convert to the desired format
+type ResultsStreamFormatParser interface {
+	// Reset the convertor to start converting a new command results
+	Reset(multiScanId, xrayVersion string, entitledForJas, multipleTargets bool) error
```

Comment:
I think "Set" is more appropriate here since we are not resetting the parser to 0 values, but rather setting them with the provided values. This may create a confusion